src/java/org/apache/cassandra/io/tries/Walker.java

```diff
@@ -57,10 +59,11 @@
     public Walker(Rebufferer source, long root)
     {
         this.source = source;
+        this.isPageAwareSupported = !(source instanceof TailOverridingRebufferer);
```

Comment:
For the production code we want to have the same behaviour regardless if `TailOverridingRebufferer` is in use. I think the right solution for this is to align the cutoff to a page boundary before writing the tail in `IncrementalTrieWriterSimple`.

Reply:
which makes simple writer more page aware - what about passing a parameter to the walker? Or create an interface, say `PageAligned` which would be the marker for rebufferers?